pex/resolver.py

```diff
@@ -806,19 +901,191 @@ def iter_targets():
           #    were specified).
           yield DistributionTarget.for_platform(platform)
 
-  resolve_request = ResolveRequest(targets=list(iter_targets()),
-                                   requirements=requirements,
-                                   requirement_files=requirement_files,
-                                   constraint_files=constraint_files,
-                                   allow_prereleases=allow_prereleases,
-                                   transitive=transitive,
-                                   indexes=indexes,
-                                   find_links=find_links,
-                                   cache=cache,
-                                   build=build,
-                                   use_wheel=use_wheel,
-                                   compile=compile,
-                                   manylinux=manylinux,
-                                   max_parallel_jobs=max_parallel_jobs)
-
-  return list(resolve_request.resolve_distributions(ignore_errors=ignore_errors))
+  download_request = DownloadRequest(
+    targets=list(iter_targets()),
+    requirements=requirements,
+    requirement_files=requirement_files,
+    constraint_files=constraint_files,
+    allow_prereleases=allow_prereleases,
+    transitive=transitive,
+    indexes=indexes,
+    find_links=find_links,
+    cache=cache,
+    build=build,
+    use_wheel=use_wheel,
+    manylinux=manylinux,
+  )
+
+  local_projects = list(download_request.iter_local_projects())
+
+  dest = dest or safe_mkdtemp()
+  download_results = download_request.download_distributions(
+    dest=dest,
+    max_parallel_jobs=max_parallel_jobs
+  )
+  return local_projects, download_results
+
+
+class LocalDistribution(namedtuple('LocalDistribution', ['target', 'path', 'fingerprint'])):
```

Comment:
The `Local` here is "has been downloaded", rather than "compatible with the local machine"?

Reply:
Correct. It's only compatible with `target` which may or may not be local.